WalletWasabi.Tests/UnitTests/Crypto/ZeroKnowledge/LinearRelationTests.cs

```diff
@@ -120,28 +99,18 @@ public void StatementAndKnowledge()
 				new Equation(b, new GroupElementVector(Generators.Gh))
 			);
 
-			// Check that simulation produces valid transcripts
 			var challenge = new Scalar(13);
-			var simulatedResponses = new[]
-			{
-				new ScalarVector(Scalar.One),
-				new ScalarVector(Scalar.One),
-			};
-
-			var simulatedPublicNonces = statement.SimulatePublicNonces(challenge, simulatedResponses);
-			Assert.True(statement.CheckVerificationEquation(simulatedPublicNonces, challenge, simulatedResponses));
 
 			// Create transcripts using a witness to the relation
 			var knowledge = new Knowledge(statement, new ScalarVector(x));
-			var allSecretNonces = new[] { new ScalarVector(new Scalar(7)), new ScalarVector(new Scalar(11)) };
-			var publicNonces = new GroupElementVector(statement.Equations.Zip(allSecretNonces, (equation, secretNonces) => secretNonces * equation.Generators));
-			var allResponses = knowledge.RespondToChallenge(challenge, allSecretNonces);
-			Assert.True(statement.CheckVerificationEquation(publicNonces, challenge, allResponses));
+			var secretNonces = new ScalarVector(new Scalar(7));
+			var publicNonces = new GroupElementVector(statement.Equations.Select(equation => secretNonces * equation.Generators));
+			var responses = Prover.RespondToChallenge(knowledge.Witness, secretNonces, challenge);
```

Comment:
Should this be named `responses` when it is an instance of `ScalarVector`. Maybe `responseVector` is clearer? Not sure.

Reply:
the plural variable names are consistently used with vector types (see also secretNonces, publicNonces, etc here), so i think if this is justified then it'd mean changing everything.